### PR TITLE
[ompl] Add vcpkg_check_linkage

### LINKED ISSUE #21658: Linux: static compilation （linux系统如何静态编译？）
platform: linux
**question:** How do I get ompl's  *.a files?(static compile)
command: (photos form installed/vcpkg/info
vcpkg install protobuf:x64-linux   => *.a
image

vcpkg install ompl:x64-linux => *.so
image

vcpkg install protobuf:arm-linux  => *.a
image

vcpkg install ompl:arm-linux => *.so
image

### PR DESCRIPTION
**Pull request by JackBoosY**
According to the [upstream configuration](https://github.com/ompl/ompl/blob/main/src/ompl/CMakeLists.txt#L49-L54):
```cmake
# build the library
if(MSVC)
    add_library(ompl STATIC ${OMPL_SOURCE_CODE})
else(MSVC)
    add_library(ompl SHARED ${OMPL_SOURCE_CODE})
endif(MSVC)
```
ompl can only be built as a static library in Windows and be built as a dynamic library in non-Windows.

Fixes #21658.

### CODE REVIEW

File: ports/ompl/portfile.cmake [hunk 1]
```diff
@@ -1,5 +1,12 @@
 vcpkg_buildpath_length_warning(37)
 
+# See https://github.com/ompl/ompl/blob/main/src/ompl/CMakeLists.txt#L49-L54
+if (VCPKG_TARGET_IS_WINDOWS)
+    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
+else()
+    vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
+endif()
```
Comment (JackBoosY) [inline ports/ompl/portfile.cmake:8]: This will not affect the dynamic/static build in Windows, because I did not add the `supports` field.

**Comment by dg0yt** [comment]
> ompl can only be built as a static library in Windows and be built as a dynamic library in non-Windows.

I wonder if this a true limitation or just convention.

File: ports/ompl/portfile.cmake [hunk 2]
```diff
@@ -1,5 +1,12 @@
 vcpkg_buildpath_length_warning(37)
 
+# See https://github.com/ompl/ompl/blob/main/src/ompl/CMakeLists.txt#L49-L54
+if (VCPKG_TARGET_IS_WINDOWS)
```
Comment (dg0yt) [inline ports/ompl/portfile.cmake:4]: `AND NOT VCPKG_TARGET_IS_MINGW`

I didn't test with mingw, but the project's configuration looks at `MSVC`.

**Review by LilyWangLL — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks :)

**Pull request merged**